src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -504,6 +569,75 @@ internal static string[] SplitWithDoubledSeparatorEscaping(string str, char sepa
                     sourceFileSeen |= optionsEnded || !IsOption(arg);
                 }
             }
+            args.Free();
+
+            void parseResponseFile(string fullPath)
+            {
+                var lines = ArrayBuilder<string>.GetInstance();
```

Comment:
I think the response file is mainly used from the msbuild task. Obviously would be nice to optimize that path too but maybe not affecting IDE as much.

Reply:
File paths in general would be worth using a `ROM` over except for the fact that every single one is eventually passed to a file system API which requires `string`. Hence forcing them to `ROM` at this layer doesn't really save anything because we always end up allocating the `string` once we get to the actual file operations.